docs/openapi-docs/src/test/resources/expected_nullable_option_class_field.yml

```diff
@@ -29,21 +29,21 @@ components:
   schemas:
     Bar:
       title: Bar
+      type: object
       required:
         - bar
-      type: object
       properties:
         bar:
           type: integer
           format: int32
     ClassWithOptionClassField:
       title: ClassWithOptionClassField
+      type: object
       required:
         - requiredStringField
-      type: object
       properties:
         optionalObjField:
-          oneOf:
+          anyOf:
```

Comment:
is this the same? oneOf vs anyOf?

Reply:
Not the same, but I believe `anyOf` is the right one to use here.
`oneOf` requires that a value matches _exactly one_ of the listed schemas, which I believe is needlessly strict may be problematic if there's an overlap between values accepted by both schemas (e.g. the referred schema is already nullable).

Reply:
So this impacts only optional object schemas? We might consider this, but I think I would move this to a separate PR, separating the "update dependency" and "change the way tapir works" changes

Reply:
This was already done in `sttp-apispec` changes (the `Schema.nullable` method which replaced the `nullable` keyword).

Reply:
https://stackoverflow.com/a/48114924/931323 supports usage of `anyOf`

Reply:
ok :) maybe we could just add a comment (probablye in apispec) as to why `anyOf` is chosen instead of `oneOf`?

Reply:
(we'll see if somebody will complain ;) )

Reply:
https://github.com/softwaremill/sttp-apispec/commit/bac957841f73e07b5f3612ea42672c48352f4474

Reply:
Thank you!

Reply:
Side note: there's a similar issue with schemas generated by `tapir` from sealed hierarchies, without the discriminator.

For example:

```scala
sealed trait Entity
case class City(name: String) extends Entity
case class Person(name: String, occupation: Option[String]) extends Entity
```

The schema for that would use `oneOf`. However, a value like `Person("Fred", None)` would serialize to JSON that would match both `City` and `Person` schemas, therefore *not* matching the overall `Entity` schema.

Using `oneOf` therefore seems incorrect in this case. Of course, probably nobody ever noticed that because in practice schemas are most likely never used for actual _validation_, they only serve to generate code and documentation.

Reply:
Yeah, it's possible that we might make a better choice there as well. But in the end, apart from sticking to the specification, it would be good to see what people actually use, so that the generated specs are understandable and useful :)

Reply:
> it would be good to see what people actually use
It's not clear to me what you mean here. We are currently generating schemas which are _technically_ incorrect. Since nobody ever complained about this, it probably doesn't make any difference for the ways that OpenAPIs are used, i.e. for generating docs and code stubs.
But that's also why I think it shouldn't hurt anybody when we change `oneOf` to `anyOf` - the tools most likely treat these in the same way.
On the other hand, this change may be more impactful for implementing things like #3645, and it would probably take off some mental load from me.

Reply:
I believe that most of the time the schemas are actually disjoint (e.g. they have a discriminator field). In that case, it's completely fine and natural to use `oneOf`, and this is probably what "the world" does in most cases. I'm only suggesting a change for non-discriminated unions, which should be rare on its own.

Reply:
Ok, thanks - we can try, and see how much of an impact this has (starting with our tests)